synapseutils/walk.py

```diff
@@ -1,6 +1,21 @@
 import synapseclient
 from synapseclient.entity import is_container
 import os
+import json
+
+def _getChildren(syn, parentId, includeTypes=["folder","file","table","link","entityview","dockerrepo"], sortBy="NAME", sortDirection="ASC", nextPageToken=None):
+    entityChildrenRequest = {'parentId':parentId,
+                             'includeTypes':includeTypes,
+                             'sortBy':sortBy,
+                             'sortDirection':sortDirection,
+                             'nextPageToken':nextPageToken}
+
+    resultPage = syn.restPOST('/entity/children',body =json.dumps(entityChildrenRequest))
+    for result in resultPage['page']:
+        yield result
+    if resultPage.get('nextPageToken') is not None:
```

Comment:
Why make this recursive?  You always run the risk of a 1000 page request which would result in a recursive limit.  Since this a very linear it is easier to write a while loop.